sirepo/pkcli/job_agent.py

```diff
@@ -253,18 +253,25 @@ def terminate(self):
         )
 
     async def _cmd(self, msg, **kwargs):
-        if msg.opName == job.OP_ANALYSIS and msg.jobCmd != 'fastcgi':
-            return await self._fastcgi_op(msg)
-        c = _Cmd
-        if msg.jobRunMode == job.SBATCH:
-            c = _SbatchRun if msg.isParallel else _SbatchCmd
-        elif msg.jobCmd == 'fastcgi':
-            c = _FastCgiCmd
-        p = c(msg=msg, dispatcher=self, op_id=msg.opId, **kwargs)
-        if msg.jobCmd == 'fastcgi':
-            self.fastcgi_cmd = p
-        self.cmds.append(p)
-        await p.start()
+        try:
+            if msg.opName == job.OP_ANALYSIS and msg.jobCmd != 'fastcgi':
+                return await self._fastcgi_op(msg)
+            c = _Cmd
+            if msg.jobRunMode == job.SBATCH:
+                c = _SbatchRun if msg.isParallel else _SbatchCmd
+            elif msg.jobCmd == 'fastcgi':
+                c = _FastCgiCmd
+            p = c(msg=msg, dispatcher=self, op_id=msg.opId, **kwargs)
+            if msg.jobCmd == 'fastcgi':
```

Comment:
for clarity, I think the `except` should be above this line. There's some global state being modified, and those operations should be atomic (or in error, i.e. exception raised). I would also move the `c` math into a function. It will clean up the code a little.